fivenines_update.sh

```diff
@@ -1,4 +1,45 @@
 #!/bin/bash
+# This script is used to update the fivenines agent, and run by root
+
+# stop the agent
+sudo systemctl stop fivenines-agent.service
+
+# if the home directory of user "fivenines" is /home/fivenines (which is the old location), migrate user's home directory to /opt/fivenines
+if [ "$(getent passwd fivenines | cut -d: -f6)" == "/home/fivenines" ]; then
+        echo "Migrating fivenines.io's working directory from /home/fivenines to /opt/fivenines"
+        # if /opt/fivenines exists, or /home/fivenines not exists, exit
+        if [ -d /opt/fivenines ] || [ ! -d /home/fivenines ]; then
+                echo "Error: /opt/fivenines already exists or /home/fivenines does not exists"
+                exit 1
+        fi
+        sudo usermod -m -d /opt/fivenines fivenines
+
+        # Reinstall the agent due to path changes
+        echo "Reinstalling the fivenines_agent package"
+        sudo su - fivenines -s /bin/bash -c 'python3 -m pipx uninstall fivenines_agent'
+        sudo su - fivenines -s /bin/bash -c 'python3 -m pipx install --index-url https://pypi.org/simple/ --pip-args="--extra-index-url https://pypi.org/simple" fivenines_agent'
+        # Check that the package is installed
+        sudo su - fivenines -s /bin/bash -c 'pipx list | grep -q fivenines_agent'
+
+        # Get the exit status of the pipx command
+        if [ $? -ne 0 ]; then
+                echo "Failed to Reinstall the fivenines_agent package"
+                exit 1
+        else
+                echo "Successfully Reinstalled the fivenines_agent package"
+        fi
+fi
+if [ "$(getent passwd fivenines | cut -d: -f6)" == "/opt/fivenines" ] && [ -d /opt/fivenines ]; then
+    echo "fivenines.io's working directory is already migrated to /opt/fivenines"
```

Comment:
This will always be printed as this is done after the migration, no ? Not sure that we need that step.

Reply:
good point